components/kyma-environment-broker/internal/process/provisioning/monitoring_integration.go

```diff
@@ -0,0 +1,111 @@
+	kebError "github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/error"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/monitoring"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+	"github.com/kyma-project/control-plane/components/provisioner/pkg/gqlschema"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	MonitoringComponentName = "rma"
+)
+
+type MonitoringIntegrationStep struct {
+	operationManager *process.ProvisionOperationManager
+	client           monitoring.Client
+	monitoringConfig monitoring.Config
+}
+
+func NewMonitoringIntegrationStep(os storage.Operations, client monitoring.Client, monitoringConfig monitoring.Config) *MonitoringIntegrationStep {
+	return &MonitoringIntegrationStep{
+		operationManager: process.NewProvisionOperationManager(os),
+		client:           client,
+		monitoringConfig: monitoringConfig,
+	}
+}
+
+func (s *MonitoringIntegrationStep) Name() string {
+	return "Monitoring_Integration"
+}
+
+func (s *MonitoringIntegrationStep) Run(operation internal.ProvisioningOperation, log logrus.FieldLogger) (internal.ProvisioningOperation, time.Duration, error) {
+	releaseName := operation.InstanceDetails.ShootName
+	isDeployed, err := s.client.IsDeployed(releaseName)
+	isPresent, err2 := s.client.IsPresent(releaseName)
+	if err != nil || err2 != nil {
+		return s.handleError(operation, err, "err while getting release", log)
+	}
+	vmPassword := monitoring.GeneratePassword(16)
+	planName, _ := broker.PlanNamesMapping[operation.ProvisioningParameters.PlanID]
+	region := ""
+	if operation.ProvisioningParameters.Parameters.Region != nil {
+		region = *operation.ProvisioningParameters.Parameters.Region
+	}
+	params := monitoring.Parameters{
+		ReleaseName:     releaseName,
+		InstanceID:      operation.InstanceID,
+		GlobalAccountID: operation.ProvisioningParameters.ErsContext.GlobalAccountID,
+		SubaccountID:    operation.InstanceDetails.SubAccountID,
+		ShootName:       operation.InstanceDetails.ShootName,
+		PlanName:        planName,
+		Region:          region,
+		Username:        operation.InstanceID,
+		Password:        vmPassword,
+	}
+
+	if !isDeployed {
+		if !isPresent {
+			log.Info("Release not found. Ready to install.")
+			_, err = s.client.InstallRelease(params)
+		} else {
+			log.Info("Release found but not deployed. Ready to upgrade.")
+			_, err = s.client.UpgradeRelease(params)
+		}
+		if err != nil {
+			return s.handleError(operation, err, "failed to deploy chart", log)
+		}
+		retry := time.Duration(0)
+		operation, retry = s.operationManager.UpdateOperation(operation, func(operation *internal.ProvisioningOperation) {
+			operation.Monitoring.Username = operation.InstanceID
+			operation.Monitoring.Password = vmPassword
```

Comment:
this value is stored in the database as a plain text. Is it a good candidate to be encrypted (some pieces of data are already being encrypted)? @ebensom ?

Reply:
I was also thinking of this, but we are going to move this step from KEB and implement it in component reconciler. So until then, it can stay as it is.